projects/views.py

```diff
@@ -148,26 +108,31 @@ def get(self, request, project_id, *args, **kwargs):
         serializer = serializer_class(page_obj, many=True)
         project_serializer = ProjectListSerializer(project_instance, many=False)
 
-        type_list = (
-            project_instance.components.all().order_by().values_list("type").distinct()
-        )
+        type_list = project_instance.components.all().order_by().values_list("type").distinct()
 
-        response = []
-        response.append({"project": project_serializer.data})
-        response.append({"components": serializer.data})
-        response.append({"total_item_count": paginator.count})
-        response.append({"type_list": type_list})
+        response = [
+            {"project": project_serializer.data},
+            {"components": serializer.data},
+            {"total_item_count": paginator.count},
+            {"type_list": type_list}
+        ]
```

Comment:
This needs a serializer so that we can utilize `generics` more, but that's out of scope of this PR.

Reply:
Would love to talk offline about this so I can understand more :)